reference, toc: add a list of expressions pushed down

fzhedu opened the pull request:
<!--Thanks for your contribution to TiDB documentation. See [CONTRIBUTING](https://github.com/pingcap/community/blob/master/CONTRIBUTING.md) before filing this PR.-->

### What is changed, added or deleted? <!--Required-->
Add a page that presents a list of expressions pushed down to TiKV

<!--Tell us what you did and why.-->
From the page, users can know which expressions have been pushed down to TiKV.

### What is the related PR or file link(s)? <!--Write "N/A" or remove this item if it is not applicable-->
N/A

<!--Provide a reference link that is related to your change. For example, a link in the pingcap/docs repository. -->

### Which version does your change affect? <!--Required; write "N/A" if it is not applicable-->
N/A

<!--Specify the version or versions that your change affect by adding a label at the right-hand side of this page. "dev" indicates the latest development version. "v3.0"/"v2.1" indicates the documentation of TiDB 3.0/2.1. If your change affects multiple versions, please update the documents for ALL the necessary versions.-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/pingcap/docs-cn?pullRequest=1802) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/pingcap/docs-cn?pullRequest=1802) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/pingcap/docs-cn?pullRequest=1802) it.</sub>

zz-jason commented:
@lilin90 @CaitinChen PTAL

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,38 @@
+category: reference
+aliases: ['/docs-cn/sql/expressions-pushed-down-2-TiKV/']
+---
+
```
```suggestion

# 下推到 TiKV 的表达式列表

```

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,38 @@
+---
+title: 下推到 TiKV 的表达式列表
+category: reference
+aliases: ['/docs-cn/sql/expressions-pushed-down-2-TiKV/']
```
`aliases` is unnecessary for a newly added file. Please remove it.

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,38 @@
+---
+
+| 表达式分类 | 具体操作 |
+| -------------- | ------------------------------------- |
```
```suggestion
| :-------------- | :------------------------------------- |
```

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,38 @@
+
+| 表达式分类 | 具体操作 |
+| -------------- | ------------------------------------- |
+| [逻辑运算](/reference/sql/functions-and-operators/operators.md) | AND (&&), OR (&#124;&#124;), NOT (!) |
```
```suggestion
| [逻辑运算](/reference/sql/functions-and-operators/operators.md#逻辑操作符) | AND (&&), OR (&#124;&#124;), NOT (!) |
```

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,38 @@
+| 表达式分类 | 具体操作 |
+| -------------- | ------------------------------------- |
+| [逻辑运算](/reference/sql/functions-and-operators/operators.md) | AND (&&), OR (&#124;&#124;), NOT (!) |
+| [比较运算](/reference/sql/functions-and-operators/operators.md) | <, <=, =, != (<>), >, >=, [`<=>`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#operator_equal-to), [`IN()`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in), IS NULL, LIKE, IS TRUE, IS FALSE, [`COALESCE()`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_co ...
```
```suggestion
| [比较运算](/reference/sql/functions-and-operators/operators.md#比较方法和操作符) | <, <=, =, != (<>), >, >=, [`<=>`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#operator_equal-to), [`IN()`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in), IS NULL, LIKE, IS TRUE, IS FALSE, [`COALESCE()`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_coalesce) |
```

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,38 @@
+| [JSON运算](/reference/sql/functions-and-operators/json-functions.md) | [JSON_TYPE(json_val)][json_type],<br> [JSON_EXTRACT(json_doc, path[, path] ...)][json_extract],<br> [JSON_UNQUOTE(json_val)][json_unquote],<br> [JSON_OBJECT(key, val[, key, val] ...)][json_object],<br> [JSON_ARRAY([val[, val] ...])][json_array],<br> [JSON_MERGE(json_doc, json_doc[, json_doc] ...)][json_merge], [JSON_SET(json_d ...
+| [日期运算](/reference/sql/functions-and-operators/date-and-time-functions.md) | [`DATE_FORMAT()`](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_date-format)  |
+
+注意：上述所支持下推的表达式可以被拉入黑名单mysql.expr_pushdown_blacklist中。
```
```suggestion
> **注意：**
>
> 上述支持下推的表达式可以被拉入黑名单 `mysql.expr_pushdown_blacklist` 中。
```

lilin90 replied:
什么时候可以或者有必要拉入黑名单呢？

lilin90 commented:
@fzhedu Does this new file also apply to `dev`, `v2.1`? If yes, please also update accordingly. Thanks!

lilin90 commented:
@fzhedu One of my comments is missed. Please check.

XuHuaiyu commented:
@lilin90 What does this error message mean?
```
error: 1 files have invalid links (or alias links which are recommended to be replaced with latest one), please fix them!

=== ERROR REPORT == :

FILE: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md
[✖] /home/circleci/pingcap/docs-cn/v3.0//reference/sql/performance/understanding-the-query-execution-plan.md
Exited with code 1
```

fzhedu commented:
> @lilin90 What does this error message mean?
> 
> ```
> error: 1 files have invalid links (or alias links which are recommended to be replaced with latest one), please fix them!
> 
> === ERROR REPORT == :
> 
> FILE: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md
> [✖] /home/circleci/pingcap/docs-cn/v3.0//reference/sql/performance/understanding-the-query-execution-plan.md
> Exited with code 1
> ```

the path is error.
 /home/circleci/pingcap/docs-cn/v3.0/reference/**sql/**performance/understanding-the-query-execution-plan.md

fzhedu commented:
> @fzhedu Does this new file also apply to `dev`, `v2.1`? If yes, please also update accordingly. Thanks

already updated related files for `2.1`, but I don't know how to update `dev`

XuHuaiyu commented on v2.1/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,141 @@
+注：
+
+1. `admin reload expr_pushdown_blacklist` 只对执行该 SQL 的 TiDB server 生效，若需要集群中所有 TiDB server 生效，需要在每台 TiDB server 上执行该 SQL。
+2. 表达式黑名单功能在 v3.0.0 及以上版本中支持。
```
2.1 do not support the expression blacklist.
So we do not need to add the related part in 2.1.

fzhedu replied:
ok, I will delete it.

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,141 @@
+2. 表达式黑名单功能在 v3.0.0 及以上版本中支持。
+3. 使用表达式原始名称文本（如，">"，"+"，"is null"）添加黑名单在 v3.0.3 及以下版本中不完全支持。v3.0.3 及以下版本中，部分表达式在黑名单中使用别名。已支持下推的表达式中，表达式别名与原始名不同的，对应关系见下表(不区分大小写)。
+
+| 表达式原始名称 | 表达式别名 | 备注 |
```
If the 3rd column is empty, is it necessary to keep it? How about removing it?

fzhedu replied:
removed.

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
````diff
@@ -0,0 +1,141 @@
+4 rows in set (0.00 sec)
+```
+
+注：
````
```suggestion
> **注意：**
```

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
````diff
@@ -0,0 +1,141 @@
+```
+
+注：
+
````
```suggestion
>
```

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,141 @@
+
+注：
+
+1. `admin reload expr_pushdown_blacklist` 只对执行该 SQL 的 TiDB server 生效，若需要集群中所有 TiDB server 生效，需要在每台 TiDB server 上执行该 SQL。
```
DO NOT use an ordered list if the order is not important.

```suggestion
> - `admin reload expr_pushdown_blacklist` 只对执行该 SQL 的 TiDB server 生效，若需要集群中所有 TiDB server 生效，需要在每台 TiDB server 上执行该 SQL。
```

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,141 @@
+注：
+
+1. `admin reload expr_pushdown_blacklist` 只对执行该 SQL 的 TiDB server 生效，若需要集群中所有 TiDB server 生效，需要在每台 TiDB server 上执行该 SQL。
+2. 表达式黑名单功能在 v3.0.0 及以上版本中支持。
```
```suggestion
> - 表达式黑名单功能在 v3.0.0 及以上版本中支持。
```

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,141 @@
+
+1. `admin reload expr_pushdown_blacklist` 只对执行该 SQL 的 TiDB server 生效，若需要集群中所有 TiDB server 生效，需要在每台 TiDB server 上执行该 SQL。
+2. 表达式黑名单功能在 v3.0.0 及以上版本中支持。
+3. 使用表达式原始名称文本（如，">"，"+"，"is null"）添加黑名单在 v3.0.3 及以下版本中不完全支持。v3.0.3 及以下版本中，部分表达式在黑名单中使用别名。已支持下推的表达式中，表达式别名与原始名不同的，对应关系见下表(不区分大小写)。
```
```suggestion
> - 使用表达式原始名称文本（如，">"，"+"，"is null"）添加黑名单在 v3.0.3 及以下版本中不完全支持。v3.0.3 及以下版本中，部分表达式在黑名单中使用别名。已支持下推的表达式中，表达式别名与原始名不同的，对应关系见下表（不区分大小写）。
```

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,141 @@
+当函数的计算过程由于下推而出现不被期待的行为时，我们通过将其拉入黑名单禁止下推来快速恢复业务。
+上述支持下推的表达式可以被拉入黑名单 `mysql.expr_pushdown_blacklist` 中，以禁止其下推。
+
+* 如何拉入黑名单？
```
Please update the indentation. It's not correct. Indent sub-items with 4 spaces. You can check this rule in our style guide.

fzhedu replied:
The indentation is added.

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,141 @@
+---
+title: 下推到 TiKV 的表达式列表
```
Add a line of `summary`.

fzhedu replied:
added

lilin90 commented on v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md:
```diff
@@ -0,0 +1,141 @@
+---
+
+# 下推到 TiKV 的表达式列表
+
```
Add a brief intro here to this whole doc.

fzhedu replied:
Added.

lilin90 commented:
@fzhedu I updated the organization and wording. Please check whether it's what you mean.

> > @fzhedu Does this new file also apply to `dev`, `v2.1`? If yes, please also update accordingly. Thanks
> 
> already updated related files for `2.1`, but I don't know how to update `dev`

- Could you please tell me where did you updated `v2.1` because I didn't see changes.
- To update `dev`, add changes to `pingcap/docs-cn/dev` like that for `v3.0`.

fzhedu commented:
> @fzhedu I updated the organization and wording. Please check whether it's what you mean.
> 
> > > @fzhedu Does this new file also apply to `dev`, `v2.1`? If yes, please also update accordingly. Thanks
> > 
> > 
> > already updated related files for `2.1`, but I don't know how to update `dev`
> 
> * Could you please tell me where did you updated `v2.1` because I didn't see changes.
> * To update `dev`, add changes to `pingcap/docs-cn/dev` like that for `v3.0`.



> @fzhedu I updated the organization and wording. Please check whether it's what you mean.
> 
> > > @fzhedu Does this new file also apply to `dev`, `v2.1`? If yes, please also update accordingly. Thanks
> > 
> > 
> > already updated related files for `2.1`, but I don't know how to update `dev`
> 
> * Could you please tell me where did you updated `v2.1` because I didn't see changes.
> * To update `dev`, add changes to `pingcap/docs-cn/dev` like that for `v3.0`.

The changes in `v2.1` is removed because the blacklist is supported since `v3.0`, which is verified by @XuHuaiyu  .
Is the version of `dev` is larger than `v3.0`?

lilin90 commented:
@fzhedu Yes. `dev` is for the future `v4.0`.
